README.md

```diff
@@ -136,59 +146,103 @@ inbound or outbound value is nil.
 If a custom type is registered for a record-type field, then any `to_avro` 
 method/Proc should return a Hash with string keys for encoding using Avro.
 
-#### Encode/Decode
+### Encoding and Decoding
+
+`Avromatic` provides two different interfaces for encoding the key (optional)
+and value associated with a model.
+
+#### Manually Managed Schemas
+
+The attributes for the value schema used to define a model can be encoded using:
+
+```ruby
+encoded_value = model.avro_raw_value
+```
+
+In order to decode this data, a copy of the value schema is required.
+
+If a model also has an Avro schema for a key, then the key attributes can be
+encoded using:
+
+```ruby
+encoded_key = model.avro_raw_key
+```
+
+If attributes were encoded using the same schema(s) used to define a model, then
+the data can be decoded to create a new model instance:
+
+```ruby
+MyTopic.raw_decode(key: encoded_key, value: encoded_value)
```

Comment:
Should we use a name like `MyModel` that is less tied to Kafka (and the assumption that topics only have a single type associated with them)?